lib/cpp/src/thrift/transport/TSocket.cpp

```diff
@@ -493,29 +585,9 @@ uint32_t TSocket::read(uint8_t* buf, uint32_t len) {
       goto try_again;
     }
 
-#if defined __FreeBSD__ || defined __MACH__
     if (errno_copy == THRIFT_ECONNRESET) {
-      /* shigin: freebsd doesn't follow POSIX semantic of recv and fails with
-       * THRIFT_ECONNRESET if peer performed shutdown
-       * edhall: eliminated close() since we do that in the destructor.
-       */
       return 0;
     }
-#endif
-
-#ifdef _WIN32
```

Comment:
Thanks Jim, I'm fine with your explanations, CMake build on appveyor currently fails and I try to get it up and runnning again. I think Windows platform users shall use Visual Studio instead of cygwin and MinGW or as a better choice clang;-)